xptifw/src/xpti_trace_framework.cpp

```diff
@@ -412,107 +397,36 @@ class Tracepoints {
-          RightPart = Payload->name_sid & Mask22Bits;
-          HashValue = LeftPart | MiddlePart | RightPart;
-          Payload->flags |=
-              static_cast<uint64_t>(payload_flag_t::HashAvailable);
-          Payload->internal = HashValue;
-          return HashValue;
-        }
-      } else if (Payload->flags &
-                 static_cast<uint64_t>(payload_flag_t::CodePointerAvailable)) {
-        // We have both kernel name and kernel address; we use bits 5-36 from
-        // the address and combine it with the kernel name string ID
-        uint32_t Address = static_cast<uint32_t>(
-            (reinterpret_cast<uint64_t>(Payload->code_ptr_va) &
-             0x0000000ffffffff0) >>
-            4);
-        HashValue = XPTI_PACK32_RET64(Address, Payload->name_sid);
-        Payload->flags |= static_cast<uint64_t>(payload_flag_t::HashAvailable);
-        Payload->internal = HashValue;
-        return HashValue;
-      } else {
-        // We only have kernel name and this is suspect if the kernel names are
-        // not unique and will replace any previously stored payload information
-        if (Payload->name_sid != xpti::invalid_id) {
-          HashValue = XPTI_PACK32_RET64(0, Payload->name_sid);
-          Payload->flags |=
-              static_cast<uint64_t>(payload_flag_t::HashAvailable);
-          Payload->internal = HashValue;
-          return HashValue;
-        }
-      }
-    } else if (Payload->flags &
-               static_cast<uint64_t>(payload_flag_t::CodePointerAvailable)) {
-      // We are only going to look at Kernel address when kernel name is not
-      // available.
-      HashValue = reinterpret_cast<uint64_t>(Payload->code_ptr_va);
-      Payload->flags |= static_cast<uint64_t>(payload_flag_t::HashAvailable);
-      Payload->internal = HashValue;
-      return HashValue;
+      // Add the kernel name to the string table
+      name_id = MStringTableRef.add(Payload->name, &Payload->name);
+    }
+    if ((Payload->flags &
+         static_cast<uint64_t>(payload_flag_t::SourceFileAvailable))) {
+      // Add source file information ot string table
+      source_id =
+          MStringTableRef.add(Payload->source_file, &Payload->source_file);
+    }
+    if ((Payload->flags &
+         static_cast<uint64_t>(payload_flag_t::StackTraceAvailable))) {
+      // Add caller/callee information ot string table
+      stack_id =
+          MStringTableRef.add(Payload->stack_trace, &Payload->stack_trace);
     }
+    // Pack the 1st 64-bit value with string ID from source file name and line
+    // number; pack the 2nd 54-bit value with stack backtrace string ID and the
+    // kernel name string ID
+    Payload->uid.p1 = XPTI_PACK32_RET64(source_id, line_no);
+    Payload->uid.p2 = XPTI_PACK32_RET64(stack_id, name_id);
+    // The code pointer for the kernel is already in 64-bit format
+    if ((Payload->flags &
+         static_cast<uint64_t>(payload_flag_t::CodePointerAvailable)))
```

Comment:
It looks like you have an extra set of parentheses here.

Reply:
Thanks! Will address in the next PR as it is cosmetic.